src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    public static final int MAX_VALUE = 11;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int randomValue = random.nextInt(MAX_VALUE);
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        switch (figureNumber) {
+            case 0: {
+                return new Circle(colorSupplier.getRandomColor(), randomValue);
+            }
+            case 1: {
+                return new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        randomValue, randomValue, randomValue);
+            }
+            case 2: {
+                return new Rectangle(colorSupplier.getRandomColor(), randomValue, randomValue);
+            }
+            case 3: {
+                return new RightTriangle(colorSupplier.getRandomColor(), randomValue, randomValue);
+            }
+            default: {
+                return new Square(colorSupplier.getRandomColor(), randomValue);
+            }
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(Color.WHITE.name(), 10);

```

Comment:
All magic numbers in your code should be constants.
Please see [this article](https://mate-academy.github.io/style-guides/java/java.html#s5.2.4-constant-names) to learn about constant fields and their naming requirements.